src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -574,6 +615,23 @@ private TNode AddOutputElement(TNode targetNode)
             return targetNode.AddElementWithCDATA("output", Output);
         }
 
+        private TNode AddAssertionsElement(TNode targetNode)
+        {
+            var assertionsNode = targetNode.AddElement("assertions");
+
+            foreach (var assertion in AssertionResults)
+            {
+                TNode assertionNode = assertionsNode.AddElement("assertion");
+                assertionNode.AddAttribute("result", assertion.Status.ToString());
+                if (assertion.Message != null)
+                    assertionNode.AddElementWithCDATA("message", assertion.Message);
+                if (assertion.StackTrace != null)
+                    assertionNode.AddElementWithCDATA("stack-trace", assertion.StackTrace);
+            }
+
+            return assertionsNode;
```

Comment:
This is the meat of all this, the XML format that will be output. The format looks fine to me. As far as I can tell, your code matches the description in the original issue, so no problems.